CHEWBBACA/CHEWBBACA_NS/down_schema.py

```diff
@@ -0,0 +1,670 @@
+"""
+Purpose
+-------
+
+This module enables the download of chewBBACA's schemas from the
+Chewie-NS.
+
+The process enables the download of ZIP archives that contain ready-to-use
+versions of any schema in the Chewie-NS. It also allows users to download
+any schema with the structure it had at a specific time point. It is also
+possible to download the latest version of the schema through requests to
+the Chewie-NS API, if the compressed version that is available does not
+match the latest version of the schema. An alternative approach that can
+be applied to get the latest version of the schema, if the compressed version
+does not provide it, is to download the compressed version that is available
+and run the SyncSchema process to retrieve the alleles that were added to the
+schema after the compression date.
+
+Expected input
+--------------
+
+The process expects the following variables wheter through command line
+execution or invocation of the :py:func:`main` function:
+
+- ``-sc``, ``schema_id`` : The schema identifier in the Chewie-NS.
+
+    - e.g.: ``1``
+
+- ``-sp``, ``species_id`` : The integer identifier or name of the species
+  that the schema will be associated to in the Chewie-NS.
+
+    - e.g.: ``1`` or ``'Yersinia pestis'``
+
+- ``-o``, ``download_folder`` : Path to the parent directory of the folder
+  that will store the downloaded schema. The process will create a folder
+  with the schema's name inside the directory specified through this argument.
+
+    - e.g.: ``/home/user/chewie_schemas``
+
+- ``--cpu``, ``cpu_cores`` : Number of CPU cores that will be used to
+  construct the schema if the process downloads FASTA files instead of
+  the compressed version.
+
+    - e.g.: ``4``
+
+- ``--ns_url``, ``nomenclature_server_url`` : The base URL for the
+  Nomenclature Server.
+
+    - e.g.: ``http://127.0.0.1:5000/NS/api/`` (local host)
+
+- ``--d``, ``date`` : Download schema with structure it had at
+  specified date. Must be in the format "Y-m-dTH:M:S" or "Y-m-dTH:M:S.f".
+
+    - e.g.: ``2020-03-27T11:38:00`` or ``2020-03-27T11:38:01.100``
+
+- ``--latest`` : If the compressed version that is available is not the
+  latest, downloads all loci and constructs schema locally.
+
+Code documentation
+------------------
+"""
+
+
+import os
+import sys
+import pickle
+import shutil
+import requests
+import argparse
+import datetime as dt
+import concurrent.futures
+from itertools import repeat
+from urllib3.exceptions import InsecureRequestWarning
+
+try:
+    from utils import constants as cnst
+    from utils import auxiliary_functions as aux
+    from PrepExternalSchema import PrepExternalSchema
+except:
+    from CHEWBBACA.utils import constants as cnst
+    from CHEWBBACA.utils import auxiliary_functions as aux
+    from CHEWBBACA.PrepExternalSchema import PrepExternalSchema
+
+
+# Suppress only the single warning from urllib3 needed.
+requests.packages.urllib3.disable_warnings(category=InsecureRequestWarning)
```

Comment:
The `InsecureRequestWarning` suppression can commented out because the warning will only be raised in local Chewie-NS instances with self-signed certificates. The same can be applied to the the warning import.